<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Data Analysis | RNAshapeQCdocs</title>
  <meta name="description" content="5 Data Analysis | RNAshapeQCdocs" />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Data Analysis | RNAshapeQCdocs" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="miyeonyeon/RNAshapeQCdocs" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Data Analysis | RNAshapeQCdocs" />
  
  
  

<meta name="author" content="Miyeon Yeon and Won-Young Choi" />


<meta name="date" content="2025-12-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="GeneBodyCoverage.html"/>
<link rel="next" href="applications.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Demo</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="data-description.html"><a href="data-description.html"><i class="fa fa-check"></i><b>2</b> Data Description</a></li>
<li class="chapter" data-level="3" data-path="data-generation.html"><a href="data-generation.html"><i class="fa fa-check"></i><b>3</b> Data Generation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-generation.html"><a href="data-generation.html#bam-slicing"><i class="fa fa-check"></i><b>3.1</b> BAM slicing</a></li>
<li class="chapter" data-level="3.2" data-path="data-generation.html"><a href="data-generation.html#pileup"><i class="fa fa-check"></i><b>3.2</b> Pileup</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="GeneBodyCoverage.html"><a href="GeneBodyCoverage.html"><i class="fa fa-check"></i><b>4</b> Data Processing</a></li>
<li class="chapter" data-level="5" data-path="data-analysis.html"><a href="data-analysis.html"><i class="fa fa-check"></i><b>5</b> Data Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-analysis.html"><a href="data-analysis.html#measures-by-rna-seq-protocol"><i class="fa fa-check"></i><b>5.1</b> Measures by RNA-seq protocol</a></li>
<li class="chapter" data-level="5.2" data-path="data-analysis.html"><a href="data-analysis.html#SOI"><i class="fa fa-check"></i><b>5.2</b> Suboptimal/Optimal index</a></li>
<li class="chapter" data-level="5.3" data-path="data-analysis.html"><a href="data-analysis.html#updated-gene-body-coverage"><i class="fa fa-check"></i><b>5.3</b> Updated gene body coverage</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>6</b> Applications</a>
<ul>
<li class="chapter" data-level="6.1" data-path="applications.html"><a href="applications.html#PCA"><i class="fa fa-check"></i><b>6.1</b> Principal component analysis</a></li>
<li class="chapter" data-level="6.2" data-path="applications.html"><a href="applications.html#window-cv-heatmap"><i class="fa fa-check"></i><b>6.2</b> Window CV heatmap</a></li>
<li class="chapter" data-level="6.3" data-path="applications.html"><a href="applications.html#sample-quality-summary"><i class="fa fa-check"></i><b>6.3</b> Sample quality summary</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">RNAshapeQCdocs</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-analysis" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">5</span> Data Analysis<a href="data-analysis.html#data-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="measures-by-rna-seq-protocol" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Measures by RNA-seq protocol<a href="data-analysis.html#measures-by-rna-seq-protocol" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figureAdd/SQmethods.png" width="85%" height="85%" style="display: block; margin: auto;" /></p>
<p>Choi et al. (2021)<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a> suggested <em>decay rate</em> by the mean-corrected slope of log-transformed data to measure the extent of degradation and identify degraded samples in <em>mRNA-seq</em>. <code>SCISSOR::decay.rate.hy()</code> calculates slopes and decay rates for all samples with exon-only coverage <code>pileupData</code>.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="data-analysis.html#cb15-1" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fu">dim</span>(pileupData)[<span class="dv">1</span>]</span>
<span id="cb15-2"><a href="data-analysis.html#cb15-2" tabindex="-1"></a>data.process <span class="ot">&lt;-</span> SCISSOR<span class="sc">::</span><span class="fu">process_pileup</span>(</span>
<span id="cb15-3"><a href="data-analysis.html#cb15-3" tabindex="-1"></a>  <span class="at">pileupData        =</span> pileupData, </span>
<span id="cb15-4"><a href="data-analysis.html#cb15-4" tabindex="-1"></a>  <span class="at">Ranges            =</span> exonRanges,</span>
<span id="cb15-5"><a href="data-analysis.html#cb15-5" tabindex="-1"></a>  <span class="at">ogshiftVal        =</span> <span class="dv">10</span>, </span>
<span id="cb15-6"><a href="data-analysis.html#cb15-6" tabindex="-1"></a>  <span class="at">plotNormalization =</span> F</span>
<span id="cb15-7"><a href="data-analysis.html#cb15-7" tabindex="-1"></a>)</span>
<span id="cb15-8"><a href="data-analysis.html#cb15-8" tabindex="-1"></a>decayRate <span class="ot">&lt;-</span> SCISSOR<span class="sc">::</span><span class="fu">decay.rate.hy</span>(<span class="at">Data=</span>data.process<span class="sc">$</span>normalizedData)<span class="sc">$</span>slope<span class="sc">*</span>d</span></code></pre></div>
<p>In case of a long gene <em>FAT1</em> with fresh frozen and mRNA-seq (FFM) 589 samples from TCGA-LUAD, total length <code>d</code> is 16,167 and <code>slope</code> is calculated by fitting a linear regression of genomic positions and the mean-corrected coverage.
In the mean-corrected coverage figure, each sample is categorized as <em>Intact</em> or <em>Degraded</em> based on the median value of <code>decayRate</code>, enabling comparison of degradation effects on base-level coverage. The degraded samples having relatively larger slopes show lower coverage.</p>
<p><img src="figureAdd/Degrate_FAT1_v001.png" width="85%" height="85%" style="display: block; margin: auto;" /></p>
<p>In our package, we focus on a novel approach based on window coefficient of variation (wCV) to assess RNA-seq data quality for <em>total RNA-seq</em>.</p>
</div>
<div id="SOI" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Suboptimal/Optimal index<a href="data-analysis.html#SOI" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A mean coverage depth (MCD) and wCV can be calculated from the <code>get_MCD()</code> and <code>get_wCV()</code> functions.
To adjust the effect of low base coverage to CV, we consider wCV within a restricted range of MCD.
Both functions return the same dimension of a matrix, which is <em>the number of genes <span class="math inline">\(\times\)</span> the number of samples</em>.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="data-analysis.html#cb16-1" tabindex="-1"></a>MCD.mat <span class="ot">=</span> <span class="fu">get_MCD</span>(</span>
<span id="cb16-2"><a href="data-analysis.html#cb16-2" tabindex="-1"></a>  <span class="at">genelist   =</span> genelist, </span>
<span id="cb16-3"><a href="data-analysis.html#cb16-3" tabindex="-1"></a>  <span class="at">pileupPath =</span> pileupPath, </span>
<span id="cb16-4"><a href="data-analysis.html#cb16-4" tabindex="-1"></a>  <span class="at">sampleInfo =</span> sampleInfo</span>
<span id="cb16-5"><a href="data-analysis.html#cb16-5" tabindex="-1"></a>)</span>
<span id="cb16-6"><a href="data-analysis.html#cb16-6" tabindex="-1"></a>wCV.mat <span class="ot">=</span> <span class="fu">get_wCV</span>(</span>
<span id="cb16-7"><a href="data-analysis.html#cb16-7" tabindex="-1"></a>  <span class="at">genelist   =</span> genelist, </span>
<span id="cb16-8"><a href="data-analysis.html#cb16-8" tabindex="-1"></a>  <span class="at">pileupPath =</span> pileupPath, </span>
<span id="cb16-9"><a href="data-analysis.html#cb16-9" tabindex="-1"></a>  <span class="at">sampleInfo =</span> sampleInfo</span>
<span id="cb16-10"><a href="data-analysis.html#cb16-10" tabindex="-1"></a>)</span></code></pre></div>
<p>The <code>get_SOI()</code> function gives the AUC of the fitted lines in the regression of wCV and log transformed MCD, and normalizes it using projection depth (PD)<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>.
Finally, we can determine the quality of the samples by defining them as <em>Suboptimal</em> if <em>PD&gt;cutoff</em>. Nineteen out of 171 samples are classified as suboptimal samples.
The SOI plot shows the distribution of AUC and PD and the shape of the regression by sample quality.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="data-analysis.html#cb17-1" tabindex="-1"></a>result <span class="ot">=</span> <span class="fu">get_SOI</span>(<span class="at">MCD=</span>MCD.mat, <span class="at">wCV=</span>wCV.mat, <span class="at">rstPct=</span><span class="dv">20</span>, <span class="at">obsPct=</span><span class="dv">50</span>)</span>
<span id="cb17-2"><a href="data-analysis.html#cb17-2" tabindex="-1"></a>auc.vec <span class="ot">&lt;-</span> result<span class="sc">$</span>auc.vec</span>
<span id="cb17-3"><a href="data-analysis.html#cb17-3" tabindex="-1"></a><span class="fu">table</span>(auc.vec<span class="sc">$</span>SOI)</span></code></pre></div>
<pre><code>## 
##    Optimal Suboptimal 
##        152         19</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="data-analysis.html#cb19-1" tabindex="-1"></a><span class="fu">plot_SOI</span>(<span class="at">SOIresult=</span>result)</span></code></pre></div>
<p><img src="plot/Fig_SOI.png" width="85%" height="85%" style="display: block; margin: auto;" /></p>
</div>
<div id="updated-gene-body-coverage" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Updated gene body coverage<a href="data-analysis.html#updated-gene-body-coverage" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The gene body coverage plot from <a href="GeneBodyCoverage.html#GeneBodyCoverage">Data Processing</a> can be updated after removing suboptimal samples using the <code>plot_GBCos()</code> function. The coverage patterns become much more stable, especially in the long genes.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="data-analysis.html#cb20-1" tabindex="-1"></a>GBCg0 <span class="ot">=</span> <span class="fu">plot_GBCos</span>(</span>
<span id="cb20-2"><a href="data-analysis.html#cb20-2" tabindex="-1"></a>  <span class="at">sampleInfo =</span> sampleInfo, </span>
<span id="cb20-3"><a href="data-analysis.html#cb20-3" tabindex="-1"></a>  <span class="at">GBCresult  =</span> GBC0, </span>
<span id="cb20-4"><a href="data-analysis.html#cb20-4" tabindex="-1"></a>  <span class="at">auc.vec    =</span> result<span class="sc">$</span>auc.vec</span>
<span id="cb20-5"><a href="data-analysis.html#cb20-5" tabindex="-1"></a>)</span>
<span id="cb20-6"><a href="data-analysis.html#cb20-6" tabindex="-1"></a>GBCg5 <span class="ot">=</span> <span class="fu">plot_GBCos</span>(</span>
<span id="cb20-7"><a href="data-analysis.html#cb20-7" tabindex="-1"></a>  <span class="at">sampleInfo =</span> sampleInfo, </span>
<span id="cb20-8"><a href="data-analysis.html#cb20-8" tabindex="-1"></a>  <span class="at">GBCresult  =</span> GBC5, </span>
<span id="cb20-9"><a href="data-analysis.html#cb20-9" tabindex="-1"></a>  <span class="at">auc.vec    =</span> result<span class="sc">$</span>auc.vec</span>
<span id="cb20-10"><a href="data-analysis.html#cb20-10" tabindex="-1"></a>)</span></code></pre></div>
<p>This function provides a continuous legend option to select such as ratio intron or PD for the line color.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="data-analysis.html#cb21-1" tabindex="-1"></a>pg0 <span class="ot">&lt;-</span> GBCg0<span class="sc">$</span>plotPD <span class="sc">+</span></span>
<span id="cb21-2"><a href="data-analysis.html#cb21-2" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.017</span>)) <span class="sc">+</span></span>
<span id="cb21-3"><a href="data-analysis.html#cb21-3" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;0~5 kb&quot;</span>)</span>
<span id="cb21-4"><a href="data-analysis.html#cb21-4" tabindex="-1"></a>pg5 <span class="ot">&lt;-</span> GBCg5<span class="sc">$</span>plotPD <span class="sc">+</span></span>
<span id="cb21-5"><a href="data-analysis.html#cb21-5" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.017</span>)) <span class="sc">+</span></span>
<span id="cb21-6"><a href="data-analysis.html#cb21-6" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;5+ kb&quot;</span>)</span>
<span id="cb21-7"><a href="data-analysis.html#cb21-7" tabindex="-1"></a></span>
<span id="cb21-8"><a href="data-analysis.html#cb21-8" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(pg0, pg5, <span class="at">common.legend=</span><span class="cn">TRUE</span>, <span class="at">legend=</span><span class="st">&quot;bottom&quot;</span>, <span class="at">nrow=</span><span class="dv">1</span>)</span></code></pre></div>
<p><img src="RNAshapeQCdocs_files/figure-html/GBCgPD-1.png" width="200%" style="display: block; margin: auto;" /></p>

</div>
</div>
<div class="footnotes">
<hr />
<ol start="2">
<li id="fn2"><p>Choi, H.Y., Jo, H., Zhao, X. et al. SCISSOR: a framework for identifying structural changes in RNA transcripts. <em>Nat Commun</em> 12, 286 (2021). <a href="https://doi.org/10.1038/s41467-020-20593-3" class="uri">https://doi.org/10.1038/s41467-020-20593-3</a><a href="data-analysis.html#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>Choi, H. (2018). Scissor for finding outliers in RNA-seq. <a href="https://doi.org/10.17615/dv2e-7a29" class="uri">https://doi.org/10.17615/dv2e-7a29</a><a href="data-analysis.html#fnref3" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="GeneBodyCoverage.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="applications.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/04-SampleQuality.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": {},
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
