<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Data Processing | RNAshapeQCdocs</title>
  <meta name="description" content="4 Data Processing | RNAshapeQCdocs" />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Data Processing | RNAshapeQCdocs" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="miyeonyeon/RNAshapeQCdocs" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Data Processing | RNAshapeQCdocs" />
  
  
  

<meta name="author" content="Miyeon Yeon and Won-Young Choi" />


<meta name="date" content="2025-12-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-generation.html"/>
<link rel="next" href="data-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Demo</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="data-description.html"><a href="data-description.html"><i class="fa fa-check"></i><b>2</b> Data Description</a></li>
<li class="chapter" data-level="3" data-path="data-generation.html"><a href="data-generation.html"><i class="fa fa-check"></i><b>3</b> Data Generation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-generation.html"><a href="data-generation.html#bam-slicing"><i class="fa fa-check"></i><b>3.1</b> BAM slicing</a></li>
<li class="chapter" data-level="3.2" data-path="data-generation.html"><a href="data-generation.html#pileup"><i class="fa fa-check"></i><b>3.2</b> Pileup</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="GeneBodyCoverage.html"><a href="GeneBodyCoverage.html"><i class="fa fa-check"></i><b>4</b> Data Processing</a></li>
<li class="chapter" data-level="5" data-path="data-analysis.html"><a href="data-analysis.html"><i class="fa fa-check"></i><b>5</b> Data Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-analysis.html"><a href="data-analysis.html#measures-by-rna-seq-protocol"><i class="fa fa-check"></i><b>5.1</b> Measures by RNA-seq protocol</a></li>
<li class="chapter" data-level="5.2" data-path="data-analysis.html"><a href="data-analysis.html#SOI"><i class="fa fa-check"></i><b>5.2</b> Suboptimal/Optimal index</a></li>
<li class="chapter" data-level="5.3" data-path="data-analysis.html"><a href="data-analysis.html#updated-gene-body-coverage"><i class="fa fa-check"></i><b>5.3</b> Updated gene body coverage</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>6</b> Applications</a>
<ul>
<li class="chapter" data-level="6.1" data-path="applications.html"><a href="applications.html#PCA"><i class="fa fa-check"></i><b>6.1</b> Principal component analysis</a></li>
<li class="chapter" data-level="6.2" data-path="applications.html"><a href="applications.html#window-cv-heatmap"><i class="fa fa-check"></i><b>6.2</b> Window CV heatmap</a></li>
<li class="chapter" data-level="6.3" data-path="applications.html"><a href="applications.html#sample-quality-summary"><i class="fa fa-check"></i><b>6.3</b> Sample quality summary</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">RNAshapeQCdocs</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="GeneBodyCoverage" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">4</span> Data Processing<a href="GeneBodyCoverage.html#GeneBodyCoverage" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The union transcript is used to extract only exon pileup. To keep only exon location, we first build coverage <code>pileup</code> from raw pileup (part_intron) to <code>pileupData</code> (only_exon).
Gene length normalization is required because gene length is a major confounder in coverage shape, raw signal intensity, and degradation sensitivity. Without length normalization, comparisons across genes or across samples become misleading.</p>
<p>In our package, gene length normalization has two options to select read depth at the region: raw value and interpolation.
After evenly dividing genomic positions by the number of regions (red vertical broken lines), suppose that both sides and center of regions are x-values of odd and even points, respectively.
While the first method finds read depth at even points (green points), and the second one finds geometric mean (blue points) using read depth at odd points (red points).
We recommend using the gene length normalization methods for gene length is at least <em>2<span class="math inline">\(\times\)</span>the number of regions+1</em> to secure minimum read depth selection across genes.</p>
<p><img src="figureAdd/MethodOption.png" width="80%" height="80%" style="display: block; margin: auto;" /></p>
<p>Before the coverage normalization, identification and filter low-expression genes need by the <code>filter_lowExpGenes()</code> function to reduce sampling noise.
Only 961 out of 1,000 genes are used for gene body coverage when considering genes for which fewer than <em>50%</em> of samples have <em>TPM &lt; 5</em>.
Genes are separated into two groups to compare base coverage patterns in short and long genes. 0~5 kb and 5+ kb cover 96 and 865 genes, respectively.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="GeneBodyCoverage.html#cb11-1" tabindex="-1"></a>pileupPath <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;../pileup/&quot;</span>, genelist, <span class="st">&quot;_pileup_part_intron.RData&quot;</span>)</span>
<span id="cb11-2"><a href="GeneBodyCoverage.html#cb11-2" tabindex="-1"></a></span>
<span id="cb11-3"><a href="GeneBodyCoverage.html#cb11-3" tabindex="-1"></a><span class="co"># Filtered genes</span></span>
<span id="cb11-4"><a href="GeneBodyCoverage.html#cb11-4" tabindex="-1"></a>genelist2 <span class="ot">&lt;-</span> <span class="fu">filter_lowExpGenes</span>(genelist, TPM, <span class="at">thr=</span><span class="dv">5</span>, <span class="at">pct=</span><span class="dv">50</span>)</span>
<span id="cb11-5"><a href="GeneBodyCoverage.html#cb11-5" tabindex="-1"></a>pileupPath2 <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;../pileup/&quot;</span>, genelist2, <span class="st">&quot;_pileup_part_intron.RData&quot;</span>)</span>
<span id="cb11-6"><a href="GeneBodyCoverage.html#cb11-6" tabindex="-1"></a></span>
<span id="cb11-7"><a href="GeneBodyCoverage.html#cb11-7" tabindex="-1"></a>geneInfo2 <span class="ot">&lt;-</span> geneInfo[<span class="fu">match</span>(genelist2, geneInfo<span class="sc">$</span>geneSymbol), ] <span class="sc">%&gt;%</span></span>
<span id="cb11-8"><a href="GeneBodyCoverage.html#cb11-8" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb11-9"><a href="GeneBodyCoverage.html#cb11-9" tabindex="-1"></a>    <span class="at">merged_kb =</span> merged<span class="sc">/</span><span class="dv">1000</span>,</span>
<span id="cb11-10"><a href="GeneBodyCoverage.html#cb11-10" tabindex="-1"></a>    <span class="at">Len       =</span> <span class="fu">factor</span>(<span class="fu">case_when</span>(</span>
<span id="cb11-11"><a href="GeneBodyCoverage.html#cb11-11" tabindex="-1"></a>      merged_kb<span class="sc">&gt;=</span><span class="dv">0</span> <span class="sc">&amp;</span> merged_kb<span class="sc">&lt;</span><span class="dv">5</span> <span class="sc">~</span> <span class="st">&quot;0~5 kb&quot;</span>,</span>
<span id="cb11-12"><a href="GeneBodyCoverage.html#cb11-12" tabindex="-1"></a>      merged_kb<span class="sc">&gt;=</span><span class="dv">5</span> <span class="sc">~</span> <span class="st">&quot;5+ kb&quot;</span>,</span>
<span id="cb11-13"><a href="GeneBodyCoverage.html#cb11-13" tabindex="-1"></a>      <span class="fu">is.na</span>(merged_kb) <span class="sc">~</span> <span class="cn">NA</span>)</span>
<span id="cb11-14"><a href="GeneBodyCoverage.html#cb11-14" tabindex="-1"></a>      ),</span>
<span id="cb11-15"><a href="GeneBodyCoverage.html#cb11-15" tabindex="-1"></a>    <span class="at">LenSorted =</span> forcats<span class="sc">::</span><span class="fu">fct_relevel</span>(Len, <span class="st">&quot;0~5 kb&quot;</span>, <span class="st">&quot;5+ kb&quot;</span>)</span>
<span id="cb11-16"><a href="GeneBodyCoverage.html#cb11-16" tabindex="-1"></a>  )</span>
<span id="cb11-17"><a href="GeneBodyCoverage.html#cb11-17" tabindex="-1"></a></span>
<span id="cb11-18"><a href="GeneBodyCoverage.html#cb11-18" tabindex="-1"></a><span class="fu">table</span>(geneInfo2<span class="sc">$</span>LenSorted)</span></code></pre></div>
<pre><code>## 
## 0~5 kb  5+ kb 
##     96    865</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="GeneBodyCoverage.html#cb13-1" tabindex="-1"></a>genelist2Len0 <span class="ot">&lt;-</span> geneInfo2[geneInfo2<span class="sc">$</span>LenSorted<span class="sc">==</span><span class="st">&quot;0~5 kb&quot;</span>, <span class="fu">c</span>(<span class="st">&quot;geneSymbol&quot;</span>)]</span>
<span id="cb13-2"><a href="GeneBodyCoverage.html#cb13-2" tabindex="-1"></a>pileupPath2Len0 <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;../pileup/&quot;</span>, genelist2Len0, <span class="st">&quot;_pileup_part_intron.RData&quot;</span>)</span>
<span id="cb13-3"><a href="GeneBodyCoverage.html#cb13-3" tabindex="-1"></a></span>
<span id="cb13-4"><a href="GeneBodyCoverage.html#cb13-4" tabindex="-1"></a>genelist2Len5 <span class="ot">&lt;-</span> geneInfo2[geneInfo2<span class="sc">$</span>LenSorted<span class="sc">==</span><span class="st">&quot;5+ kb&quot;</span>, <span class="fu">c</span>(<span class="st">&quot;geneSymbol&quot;</span>)]</span>
<span id="cb13-5"><a href="GeneBodyCoverage.html#cb13-5" tabindex="-1"></a>pileupPath2Len5 <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;../pileup/&quot;</span>, genelist2Len5, <span class="st">&quot;_pileup_part_intron.RData&quot;</span>)</span></code></pre></div>
<p>In the <code>plot_GBC()</code> function, evenly spaced regions are defined as gene body percentile where the number of regions is 100.
Unstable patterns in base coverage especially in short genes are still detected even after filtering low-expression genes.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="GeneBodyCoverage.html#cb14-1" tabindex="-1"></a>GBC0 <span class="ot">=</span> <span class="fu">plot_GBC</span>(</span>
<span id="cb14-2"><a href="GeneBodyCoverage.html#cb14-2" tabindex="-1"></a>  <span class="at">pileupPath =</span> pileupPath2Len0, </span>
<span id="cb14-3"><a href="GeneBodyCoverage.html#cb14-3" tabindex="-1"></a>  <span class="at">geneNames  =</span> genelist2Len0, </span>
<span id="cb14-4"><a href="GeneBodyCoverage.html#cb14-4" tabindex="-1"></a>  <span class="at">sampleInfo =</span> sampleInfo</span>
<span id="cb14-5"><a href="GeneBodyCoverage.html#cb14-5" tabindex="-1"></a>)</span>
<span id="cb14-6"><a href="GeneBodyCoverage.html#cb14-6" tabindex="-1"></a>GBC5 <span class="ot">=</span> <span class="fu">plot_GBC</span>(</span>
<span id="cb14-7"><a href="GeneBodyCoverage.html#cb14-7" tabindex="-1"></a>  <span class="at">pileupPath =</span> pileupPath2Len5, </span>
<span id="cb14-8"><a href="GeneBodyCoverage.html#cb14-8" tabindex="-1"></a>  <span class="at">geneNames  =</span> genelist2Len5, </span>
<span id="cb14-9"><a href="GeneBodyCoverage.html#cb14-9" tabindex="-1"></a>  <span class="at">sampleInfo =</span> sampleInfo</span>
<span id="cb14-10"><a href="GeneBodyCoverage.html#cb14-10" tabindex="-1"></a>)</span>
<span id="cb14-11"><a href="GeneBodyCoverage.html#cb14-11" tabindex="-1"></a></span>
<span id="cb14-12"><a href="GeneBodyCoverage.html#cb14-12" tabindex="-1"></a>p0 <span class="ot">&lt;-</span> GBC0<span class="sc">$</span>plot <span class="sc">+</span></span>
<span id="cb14-13"><a href="GeneBodyCoverage.html#cb14-13" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.017</span>)) <span class="sc">+</span></span>
<span id="cb14-14"><a href="GeneBodyCoverage.html#cb14-14" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;0~5 kb&quot;</span>)</span>
<span id="cb14-15"><a href="GeneBodyCoverage.html#cb14-15" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> GBC5<span class="sc">$</span>plot <span class="sc">+</span></span>
<span id="cb14-16"><a href="GeneBodyCoverage.html#cb14-16" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.017</span>)) <span class="sc">+</span></span>
<span id="cb14-17"><a href="GeneBodyCoverage.html#cb14-17" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;5+ kb&quot;</span>)</span>
<span id="cb14-18"><a href="GeneBodyCoverage.html#cb14-18" tabindex="-1"></a></span>
<span id="cb14-19"><a href="GeneBodyCoverage.html#cb14-19" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(p0, p5, <span class="at">common.legend=</span><span class="cn">TRUE</span>, <span class="at">legend=</span><span class="st">&quot;bottom&quot;</span>, <span class="at">nrow=</span><span class="dv">1</span>)</span></code></pre></div>
<p><img src="figureAdd/GBC-1.png" width="90%" height="90%" style="display: block; margin: auto;" /></p>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-generation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="data-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/03-DataProcessing.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": {},
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
